tests/base/src/test/java/org/wildfly/security/auth/realm/DistributedSecurityRealmTest.java

```diff
@@ -126,7 +136,43 @@ public void testBadEvidence() throws Exception {
         identity.dispose();
     }
 
+    @Test(expected = RealmUnavailableException.class)
+    public void testRealmUnavailable() throws Exception {
+        realm = new DistributedSecurityRealm(realm1, realm2, unavailableLdapRealm, realm3);
 
+        realm.getRealmIdentity(new NamePrincipal("user3"));
+    }
+
+    @Test(expected = RealmUnavailableException.class)
+    public void testRealmUnavailableIgnoreFalse() throws Exception {
+        boolean ignoreUnavailableRealms = false;
+        realm = new DistributedSecurityRealm(ignoreUnavailableRealms, null, realm1, realm2, unavailableLdapRealm, realm3);
+
+        realm.getRealmIdentity(new NamePrincipal("user3"));
+    }
+
+    @Test
+    public void testIgnoreRealmUnavailable() throws Exception {
+        boolean ignoreUnavailableRealms = true;
+        realm = new DistributedSecurityRealm(ignoreUnavailableRealms, null, realm1, realm2, unavailableLdapRealm, realm3);
+
+        RealmIdentity identity = realm.getRealmIdentity(new NamePrincipal("user3"));
+        Assert.assertTrue(identity.exists());
+        identity.dispose();
+    }
+
+    @Test
+    public void testIgnoreRealmUnavailableWithConsumer() throws Exception {
+        int[] failedRealmIndex = {-1};
+
+        boolean ignoreUnavailableRealms = true;
+        realm = new DistributedSecurityRealm(ignoreUnavailableRealms, (index) -> failedRealmIndex[0] = index, realm1, realm2, unavailableLdapRealm, realm3);
+
+        RealmIdentity identity = realm.getRealmIdentity(new NamePrincipal("user3"));
+        Assert.assertTrue(identity.exists());
+        Assert.assertEquals(2 , failedRealmIndex[0]);
+        identity.dispose();
+    }
```

Comment:
It would be good to add a test for the case where `realm.getRealmIdentity()` throws a `RealmUnavailableException`. See `UnavailableRealm` in `FailoverRealmTest` for an example.